jung-visualization/src/main/java/edu/uci/ics/jung/visualization/BasicVisualizationServer.java

```diff
@@ -542,23 +536,9 @@ public void paint(Graphics g) {
                 .inverseTransform(new Point2D.Double(r.getX(), r.getY()));
         String label = num + ":" + ((SpatialGrid) spatialGrid).getMap().get(num);
         int stringWidth = g2d.getFontMetrics().stringWidth(label);
-        Shape shape = r;
-        if (viewTransformer instanceof LensTransformer) {
-          shape = multiLayerTransformer.transform(shape);
-        } else if (layoutTransformer instanceof LensTransformer) {
-          LayoutModel<N, Point2D> layoutModel = model.getLayoutModel();
-          Dimension d = new Dimension(layoutModel.getWidth(), layoutModel.getHeight());
-
-          HyperbolicShapeTransformer shapeChanger =
-              new HyperbolicShapeTransformer(d, viewTransformer);
-          LensTransformer lensTransformer = (LensTransformer) layoutTransformer;
-          shapeChanger.getLens().setLensShape(lensTransformer.getLens().getLensShape());
-          MutableTransformer layoutDelegate =
-              ((MutableTransformerDecorator) layoutTransformer).getDelegate();
-          shape = shapeChanger.transform(layoutDelegate.transform(shape));
-        } else {
-          shape = getRenderContext().getMultiLayerTransformer().transform(Layer.LAYOUT, shape);
-        }
+
```

Comment:
Helpful comment: I factored out all the lens messy-ness into TransformSupport and LensTransformSupport to clean up this class